### PR TITLE
Mac Fixes

### PR DESCRIPTION
**Pull request by melMass**
I cannot check other OSes right now to see if it breaks but those modifications work on mac.
I split it in 1 commit per line

Let me know

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -14,7 +14,7 @@ link_directories( DemoGafferExtension ${GAFFER_ROOT}/lib )
 add_library( DemoGafferExtension SHARED ${DemoGafferExtensionSrc} )
 target_compile_definitions( DemoGafferExtension PRIVATE BOOST_SIGNALS_NO_DEPRECATION_WARNING=1 )
 target_include_directories( DemoGafferExtension PRIVATE include ${GAFFER_ROOT}/include )
-target_link_libraries( DemoGafferExtension Gaffer GafferScene )
+target_link_libraries( DemoGafferExtension Gaffer GafferScene IECore IECorePython boost_python Python)
```
Comment (andrewkaufman) [inline CMakeLists.txt]: Adding IECore is fine here, but the rest should be added below in the python module (line 27). We don’t want the base lib linking to python.

File: CMakeLists.txt [hunk 2]
```diff
@@ -4,13 +4,17 @@ project(DemoGafferExtension)
 set(CMAKE_CXX_STANDARD 11)
 set(CMAKE_VERBOSE_MAKEFILE OFF)
 
+if(APPLE)
```
Comment (andrewkaufman) [inline CMakeLists.txt:7]: I think it’s more common to use DARWIN

File: CMakeLists.txt [hunk 3]
```diff
@@ -19,7 +23,7 @@ add_library( DemoGafferExtensionModule SHARED ${DemoGafferExtensionModuleSrc} )
 set_target_properties( DemoGafferExtensionModule PROPERTIES PREFIX "" OUTPUT_NAME "_DemoGafferExtension" )
 target_compile_definitions( DemoGafferExtensionModule PRIVATE BOOST_SIGNALS_NO_DEPRECATION_WARNING=1 )
 target_link_libraries( DemoGafferExtensionModule DemoGafferExtension )
-target_include_directories( DemoGafferExtensionModule PRIVATE include ${GAFFER_ROOT}/include ${GAFFER_ROOT}/include/python2.7 )
```
Comment (andrewkaufman) [inline CMakeLists.txt:22]: This should have an if/else block testing for DARWIN

**Review by andrewkaufman — COMMENTED** [review]
Thanks

**Comment by melMass** [comment]
I've made all modifications from comments. Darwin returns false on both my stations so I've used `APPLE` again. Not sure why.

File: CMakeLists.txt (hunk 2 above)
Comment (melMass) [reply CMakeLists.txt:7]: Not on my end

Comment (andrewkaufman) [reply CMakeLists.txt:7]: oh, ok well if APPLE works lets stick with that

File: CMakeLists.txt [hunk 4]
```diff
@@ -23,7 +23,13 @@ add_library( DemoGafferExtensionModule SHARED ${DemoGafferExtensionModuleSrc} )
 set_target_properties( DemoGafferExtensionModule PROPERTIES PREFIX "" OUTPUT_NAME "_DemoGafferExtension" )
 target_compile_definitions( DemoGafferExtensionModule PRIVATE BOOST_SIGNALS_NO_DEPRECATION_WARNING=1 )
 target_link_libraries( DemoGafferExtensionModule DemoGafferExtension )
-target_include_directories( DemoGafferExtensionModule PRIVATE include ${GAFFER_ROOT}/include ${GAFFER_ROOT}/lib/Python.framework/Versions/2.7/include/python2.7 )
+
+if(DARWIN)
+  target_include_directories( DemoGafferExtensionModule PRIVATE include ${GAFFER_ROOT}/include ${GAFFER_ROOT}/lib/Python.framework/Versions/2.7/include/python2.7 )
+else()
+  target_include_directories( DemoGafferExtensionModule PRIVATE include ${GAFFER_ROOT}/include ${GAFFER_ROOT}/include/python2.7 )
+endif(DARWIN)
```
Comment (andrewkaufman) [inline CMakeLists.txt]: I'm not super familiar with CMake, but is this the right syntax for closing out an `if`? I'm just surprised to see we need to re-specify the initial condition...

**Comment by andrewkaufman** [comment]
Thanks, can you squash it all down into one commit?

File: CMakeLists.txt (hunk 4 above)
Comment (melMass) [reply CMakeLists.txt]: Not mandatory at all but it can help on concatenated statements. Mainly for clarity sake.

**Comment by melMass** [comment]
@andrewkaufman you can do precisely that when merging. It's actually called squash I think:

- Merge: Merge all commits from pull request at the tail of master
- Rebase: Find the latest common commit an insert pull request commits on master.
- Squash (I think): Merge all modifications from pull to one single commit (what you want.)

**Pull request merged**